src/utils/twoFactor.js

```diff
@@ -0,0 +1,181 @@
+import * as nearApiJs from 'near-api-js'
+import { store } from '..'
+import { promptTwoFactor } from '../actions/account'
+import { ACCESS_KEY_FUNDING_AMOUNT, splitPK, toPK } from './wallet'
+
+export const METHOD_NAMES_LAK = ['add_request', 'add_request_and_confirm', 'delete_request', 'confirm']
+const VIEW_METHODS = ['get_request_nonce', 'list_request_ids']
+const METHOD_NAMES_CONFIRM = ['confirm']
+/********************************
+Helpers
+********************************/
+const deleteUnconfirmedRequests = async (contract) => {
+    const request_ids = await contract.list_request_ids().catch((e) => { console.log(e) })
+    if (!request_ids || request_ids.length === 0) {
+        return
+    }
+    console.log(request_ids)
+    const promises = []
+    // try to unconfirmed requests using current pk, catch exceptions, fail fast so other promises can run
+    for (let i = 0; i < request_ids.length; i++) {
+        promises.push(contract.delete_request({ request_id: request_ids[i] }).catch((e) => console.log(e)))
+    }
+    try {
+        await Promise.all(promises)
+    } catch (e) {
+        // take no action if request cannot be deleted (probably due to cooldown period for new request)
+        console.log(e)
+    }
+}
+const getNextRequestId = async (contract) => {
+    return contract.get_request_nonce().catch((e) => { console.log(e) })
```

Comment:
why ignore error? why `.catch` in `async` function?

Reply:
updated to try, catch